Blizzard.IO.RabbitMQ/Rpc/AbstractMessageSerializationStrategy.cs

```diff
@@ -0,0 +1,35 @@
+﻿using EasyNetQ;
+using System;
+using ISerializer = Blizzard.IO.Core.Rpc.ISerializer;
+
+namespace Blizzard.IO.RabbitMQ.Rpc
+{
+    public class AbstractMessageSerializationStrategy : IMessageSerializationStrategy
+    {
+        private readonly ISerializer _serializer;
+        private readonly ITypeNameSerializer _typeNameSerializer;
+
+        public AbstractMessageSerializationStrategy(ISerializer serializer, ITypeNameSerializer typeNameSerializer)
+        {
+            _serializer = serializer;
+            _typeNameSerializer = typeNameSerializer;
+        }
```

Comment:
On our team its a convention to use:
`_someDependency = someDependency ?? throw new ArgumentNullException(nameof(someDependency))`
In order to avoid any case of `null` injection in the constructor which will prevent the class from functioning
I personally think that it can be useful, what do you think?

Reply:
You are right about this one and i also wanted to comment it, specifically here it really is redundant. 
I guess we could neglect this practice in our project :)

Reply:
Or maybe do it only in classes the user is expected to construct